### ISSUE TITLE
`Gopkg.lock` changes when running `dep ensure`

### ISSUE THREAD

**Issue by cweagans** [drud/ddev#548]
## What happened (or feature request):

If I clone this repo, then run `dep ensure`, I see this change to `Gopkg.lock`:

```
diff --git a/Gopkg.lock b/Gopkg.lock
index 0a4ee673..afc162d5 100644
--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -74,6 +74,12 @@
   version = "v0.3.2"
 
 [[projects]]
+  name = "github.com/drud/ddev"
+  packages = ["cmd/ddev/cmd","pkg/appimport","pkg/appports","pkg/archive","pkg/cms/config","pkg/cms/model","pkg/ddevapp","pkg/dockerutil","pkg/exec","pkg/fileutil","pkg/output","pkg/plugins/platform","pkg/testco
+  revision = "99a18b5e51efe0f86284f11ed2d8771710432415"
+  version = "v0.10.0"
+
+[[projects]]
   branch = "master"
   name = "github.com/drud/go-pantheon"
   packages = ["pkg/pantheon"]
@@ -322,6 +328,6 @@
 [solve-meta]
   analyzer-name = "dep"
   analyzer-version = 1
-  inputs-digest = "25c7be4cfa247fa46582ee64ea5cae9c66b43af42d26f935b04bcc93a344a5da"
+  inputs-digest = "7e0ee1003fca1435ff459b18a44828cfc795b19566f4494c76299b11e8592389"
   solver-name = "gps-cdcl"
   solver-version = 1
```

I also see a substantial number of changes to the `vendor/` directory.

## What you expected to happen:

`Gopkg.lock` shouldn't change unless `Gopkg.toml` changes, or if project dependency usage changes.

## How to reproduce this:

1. Clone the repo
2. Run `dep ensure`

## Version: Please include the output of `ddev version`, `docker version` and the project's .ddev/config.yaml.

* `dep` version: 0.3.2

## Anything else do we need to know:

* We should consider removing the `vendor/` dir from the repository.

## Related source links or issues:

* https://github.com/golang/dep/issues/1392#issuecomment-346305224
* https://github.com/golang/dep/blob/master/docs/FAQ.md#should-i-commit-my-vendor-directory

## Please use a complexity rating of 1-5 (5 is high) for a feature request. (High complexity implies more PR planning)

1

**Comment by cweagans** [comment]
`dep prune` gets rid of the vendor changes, but the lock file should still not change.

**Comment by cweagans** [comment]
I had the repo cloned in the wrong place. Now, the behavior is consistent with @tannerjfco's system:

```
diff --git a/Gopkg.lock b/Gopkg.lock
index 0a4ee673..758b9c33 100644
--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -322,6 +322,6 @@
 [solve-meta]
   analyzer-name = "dep"
   analyzer-version = 1
-  inputs-digest = "25c7be4cfa247fa46582ee64ea5cae9c66b43af42d26f935b04bcc93a344a5da"
+  inputs-digest = "184fa914c923f63ed78adaf9f77216e2cf07a0650691ffaba1d9277084b55750"
   solver-name = "gps-cdcl"
   solver-version = 1
```

I still don't think this should be changing and we should consider removing `vendor/`, but there are fewer changes at least.

**Comment by rfay** [comment]
From https://github.com/golang/dep/blob/master/README.md:

> There is one main subcommand you will use: dep ensure. ensure first checks that Gopkg.lock is consistent with Gopkg.toml and the imports in your code. If any changes are detected, dep's solver works out a new Gopkg.lock. Then, dep checks if the contents of vendor/ are what Gopkg.lock (the new one if applicable, else the existing one) says it should be, and rewrites vendor/ as needed to bring it into line.

One other little detail: We use `dep ensure` followed by `dep prune`. However, the prune behavior is (soon) to be included in ensure.

So I think this isn't a bug at this time. When dep stabilizes, and we like using it all the time, we can consider removing vendor. Remember that this is an area of phenomenal immaturity in golang.

**Comment by rfay** [comment]
I think by the time https://github.com/drud/ddev/pull/661 lands, this can be closed. Pretty sure this ends up actually being a dup of #661

**Comment by rfay** [comment]
Pretty sure this was fixed in #661, closing. Please reopen if not.
